### DISCUSSION TITLE
Having issue with inheritance from classes with default values with MappedAsDataclass and pydantic.dataclasses

### DISCUSSION THREAD

**Discussion by oliviernguyenquoc** [sqlalchemy/sqlalchemy#9994] [Usage Questions] [answered]
### Describe the bug

I can't inheritated a Parent class with default attributes in a Child class with non-default attributes.
(Please look at the parent class, it's specific to SQLAlchemy Declarative mapping style using pydantic)

I already tried to add `@dataclass(kw_only=True)` to my child class, without sucess

### SQLAlchemy Version in Use

2.0.16

### DBAPI (i.e. the database driver)

psycopg

### Database Vendor and Major Version

PostgreSQL

### Python Version

3.11

### Operating system

OSX

### To Reproduce

I defined a Base class:

```python
from datetime import datetime

import pydantic.dataclasses
from pydantic.dataclasses import dataclass
from sqlalchemy import DateTime
from sqlalchemy.orm import DeclarativeBase, Mapped, MappedAsDataclass, mapped_column


class Base(
    MappedAsDataclass,
    DeclarativeBase,
    dataclass_callable=pydantic.dataclasses.dataclass,
):
    class Config:
        orm_mode = True


class DefaultColumnMixin(MappedAsDataclass):
    id: Mapped[int] = mapped_column(primary_key=True)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        default=datetime.now(),
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), default=datetime.now(), onupdate=datetime.now()
    )
```

Inheritated a from these classes:
```python

from pydantic.dataclasses import dataclass
from sqlalchemy import String
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base, DefaultColumnMixin


class User(Base, DefaultColumnMixin):
    __tablename__ = "users"

    name: Mapped[str] = mapped_column(String(50))
```

### Error

```
sqlalchemy.exc.InvalidRequestError: Python dataclasses error encountered when creating dataclass for 'Brand': TypeError("non-default argument 'name' follows default argument"). Please refer to Python dataclasses documentation for additional information. (Background on this error at: https://sqlalche.me/e/20/dcte)
```

**Comment by zzzeek** [comment, marked as answer]
we dont support pydantic right now and the 1.x series of pydantic is known to have many problems with SQLAlchemy.  Try out the 2.x branch of pydantic for now.

**Reply by oliviernguyenquoc** [reply]
Oh you are right !
Seems to work like a charm in pydantic 2.X
It wasn't that obvious from the doc (but it will not be important in a near future with the 2.0 pydantic official release)

Still having a pylance warning but it maybe corrected on Pylance side

**Reply by zzzeek** [reply]
great that 2.x works
